Symfony/CS/Fixer/Contrib/ClassyDefinitionFixer.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class ClassyDefinitionFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->getSize() - 1; $index > 0;--$index) {
+            if (!$tokens[$index]->isClassy()) {
+                continue;
+            }
+            $classyStart = $tokens->getNextTokenOfKind($index, array('{'));
+            for ($j = $index + 1; $j < $classyStart; ++$j) {
+                if ($tokens[$j]->isWhitespace()) {
+                    $content = $tokens[$j]->getContent();
+                    if (false === $breakAt = strpos($content, "\n")) {
```

Comment:
>inline assignment and yoda together are bad, each for itself already is. It would be better to put the assignment on a line itself above.

Can you elaborate on that?
I think yoda-style is common practice in some projects (including Symfony http://symfony.com/doc/current/contributing/code/standards.html), so why would it be `bad`?
How would putting the declaration above it improve something? Is it readability or is there some technical gain as well?